### ISSUE TITLE
Favour chain Id over genesis state hash to add network uniqueness to the staking ledgers persistence

### ISSUE THREAD

**Issue by trevorbernard** [Granola-Team/mina-indexer#1222]
Currently we append the genesis state hash to uniquely identify the staking ledgers but better would be to use the current chainId.

**Comment by robinbb** [comment]
Did the chainId change upon hard fork?

**Comment by trevorbernard** [comment]
> Did the chainId change upon hard fork?

Yes, it's generated from the networks genesis constants and a timestamp.

Also, I believe that's how MinaExplorer and the Mina Payout Scripts handle it

**Comment by jhult** [comment]
Related to #1611

**Comment by jhult** [comment]
Related to #1768

**Comment by jhult** [comment]
@Isaac-DeFrain and @n1tranquilla - is this still necessary/desired?

**Comment by Isaac-DeFrain** [comment]
We'll keep using the genesis state hashes. We know the chain id if we know the genesis state hash.
